### PR TITLE
Added training method that accepts initial predictor for Symboli SGD estimator

### LINKED ISSUE #1087: Symbolic SGD should support training from initial predictor
In the estimator conversion, the functionality of training with weights taken from a previously trained predictor was lost. It should be restored. This was identified by Tom in his comment: 

https://github.com/dotnet/machinelearning/pull/1012/files/f7f666615e8befc0cd8b6925efe0a4c1c5e14101#r220340515

### PR DESCRIPTION
**Pull request by artidoro**
Fixes #1087.

This PR restores the ability to train SymbolicSGD estimator starting from the weights of a previously trained predictor.

I added a  test to demonstrate that it works.

### CODE REVIEW

File: test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs [hunk 1]
```diff
@@ -20,5 +26,42 @@ public void TestEstimatorSymSgdClassificationTrainer()
...
+            var col2 = outInitData.GetColumn<float>(Env, "Score").Take(numExamples).ToArray();
+            var col3 = outNoInitData.GetColumn<float>(Env, "Score").Take(numExamples).ToArray();
+
+            bool col12Diff = default;
+            bool col23Diff = default;
+            bool col13Diff = default;
+
+            for (int i = 0; i < numExamples; i++)
+            {
+                Console.WriteLine(col1[i].ToString() + " " + col2[i].ToString() + " " + col3[i].ToString());
```
Comment (sfilipi) [inline test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs]: > Console.WriteLine(col1[i [](start = 15, length = 25)

do you need it in console, or file?

File: test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs [hunk 2]
```diff
@@ -20,5 +26,42 @@ public void TestEstimatorSymSgdClassificationTrainer()
...
+
+            var notInitPredictor = new SymSgdClassificationTrainer(Env, "Features", "Label").Train(transformedData);
+            var outNoInitData = notInitPredictor.Transform(transformedData);
+
+            int numExamples = 10;
+            var col1 = data.GetColumn<float>(Env, "Score").Take(numExamples).ToArray();
+            var col2 = outInitData.GetColumn<float>(Env, "Score").Take(numExamples).ToArray();
+            var col3 = outNoInitData.GetColumn<float>(Env, "Score").Take(numExamples).ToArray();
+
+            bool col12Diff = default;
```
Comment (sfilipi) [inline test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs]: you can also omit it for primitives.

File: test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs (hunk 1 above)
Comment (artidoro) [reply test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs]: Sorry that was just meant for debugging, I will remove that.

File: test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs (hunk 2 above)
Comment (artidoro) [reply test/Microsoft.ML.Tests/TrainerEstimators/SymSgdClassificationTests.cs]: For some reason I then get warned that I use a unassigned variable.

**Pull request merged**